torch/csrc/jit/script/compiler.cpp

```diff
@@ -1116,26 +1137,21 @@ struct to_ir {
 
   Value* emitCond(const Expr& cond) {
     Value* v = emitExpr(cond);
-    if (!v->type()->isSubtypeOf(BoolType::get())) {
-      Value* cast_v = emitBuiltinCall(
-          cond.get()->range(),
-          *v->owningGraph(),
-          prim::Bool,
-          c10::nullopt,
-          {v},
-          {},
-          /*required*/ false);
-      if (cast_v == nullptr) {
-        ErrorReport error(cond);
-        error
-            << "expected a bool, int, float, or Tensor expression for condition but found "
-            << v->type()->str();
-        throw error;
-      } else {
-        v = cast_v;
-      }
+    Value* out;
+    try {
```

Comment:
Why is this try-catch necessary? Shouldn't `bool_cast` be a CastValue and throw the right error?

Reply:
It's to report a better error message. Otherwise it would just show prim::Bool schema matching errors. I guess i could move this whole logic to CastValue.